### ISSUE TITLE
TimeSensor should allow a description field

### ISSUE THREAD

**Issue by coderextreme** [castle-engine/view3dscene#67]
Per:

https://www.web3d.org/specifications/X3Dv4Draft/ISO-IEC19775-1v4-IS.proof/Part01/components/time.html#TimeSensor

Errors thrown from tovrmlx3d.exe:

$ ~/Downloads/view3dscene-4.3.0-win64-x86_64/view3dscene/tovrmlx3d.exe --validate lily_73.gltfholger.x3d
tovrmlx3d: Warning: X3D: Unknown X3D field name (unhandled X3D XML attribute) "description" in node "TimeSensor"
tovrmlx3d: Warning: X3D: Unknown X3D field name (unhandled X3D XML attribute) "description" in node "TimeSensor"
Exception "Exception":
Validation failed (consult the warnings above), exiting with non-zero status

$ ~/Downloads/view3dscene-4.3.0-win64-x86_64/view3dscene/tovrmlx3d.exe --validate lily_73.gltfholger.x3dv
tovrmlx3d: Warning: VRML/X3D: Error when reading, will skip the rest of X3D file: Error at line 296901 column 22: Invalid X3D node content (probably unknown or not allowed field, prototype or VRML 1.0-style children) inside "TimeSensor": got "description"
Exception "Exception":
Validation failed (consult the warnings above), exiting with non-zero status

$ ls -ld ~/Downloads/view3dscene-4.3.0-win64-x86_64
drwxr-xr-x 1 john 197609 0 Sep 17 01:24 /c/Users/john/Downloads/view3dscene-4.3.0-win64-x86_64/

My workaround that is planned is to remove the description for now (this is a generated file).

Also note that view3dscene really balloons the size of a glTF (over 2 orders of magnitude) when converting to XML and VRML. I'm not sure what's up at this point.  glTF may be a better solution, but I like my IndexedFaceSet! I'll test with the new copy.

Um, wow!

$ ls -ltr lily_73.gltf lily_73.gltfsource.x3dv
-rw-r--r-- 1 john 197609   2131969 Sep 11 11:41 lily_73.gltf
-rw-r--r-- 1 john 197609 786760983 Sep 17 01:36 lily_73.gltfsource.x3dv

**Comment by michaliskambi** [comment]
Added `TimeSensor.description` in https://github.com/castle-engine/castle-engine/commit/2a0d2e8fdafbfa7cd455d0cec5975c75bb834ad7 .

> Also note that view3dscene really balloons the size of a glTF (over 2 orders of magnitude) when converting to XML and VRML. I'm not sure what's up at this point. glTF may be a better solution, but I like my IndexedFaceSet! I'll test with the new copy.

We talked about this on x3d-public in the past. X3D encodings (classic and XML) store all numbers (including huge number of per-vertex data) as text. While in glTF, they are stored in binary blob, which is both much faster to read (no parsing), perfectly precise, and better for file size.
